drivers/regulator/regulator_gpio.c

```diff
@@ -0,0 +1,177 @@
+
+	return cfg->states_cnt;
+}
+
+static int regulator_gpio_list_voltage(const struct device *dev, unsigned int idx,
+				       int32_t *volt_uv)
+{
+	const struct regulator_gpio_config *cfg = dev->config;
+
+	if (idx >= cfg->states_cnt) {
+		LOG_ERR("%s: can't get list voltage for idx %u", dev->name, idx);
+		return -EINVAL;
+	}
+
+	*volt_uv = cfg->states[idx * 2];
+	return 0;
+}
+
+static int regulator_gpio_set_voltage(const struct device *dev, int32_t min_uv, int32_t max_uv)
+{
+	const struct regulator_gpio_config *cfg = dev->config;
+	struct regulator_gpio_data *data = dev->data;
+	int32_t best_voltage = INT32_MAX;
+	unsigned int best_state;
+	int ret = 0;
+
+	/* choose minimum possible voltage in range provided by a caller */
+	for (unsigned int state = 0; state < cfg->states_cnt; state++) {
+		if (!IN_RANGE(cfg->states[state * 2], min_uv, max_uv) ||
+		    cfg->states[state * 2] >= best_voltage) {
+			continue;
+		}
+
+		best_voltage = cfg->states[state * 2];
+		best_state = state;
+	}
+
+	if (best_voltage == INT32_MAX) {
+		LOG_ERR("%s: can't find voltage is states", dev->name);
+		return -EINVAL;
+	}
+
+	ret = gpio_pin_get_dt(&cfg->switch_volt);
+	if (ret < 0) {
+		LOG_ERR("%s: can't get pin state", dev->name);
+		return ret;
+	}
+
+	if (ret != cfg->states[best_state * 2 + 1]) {
+		ret = gpio_pin_set_dt(&cfg->switch_volt, cfg->states[best_state * 2 + 1]);
+		if (ret < 0) {
+			LOG_ERR("%s: can't set pin state", dev->name);
+			return ret;
+		}
+	}
+
+	data->current_volt_uv = cfg->states[best_state * 2];
+	return 0;
+}
+
+static int regulator_gpio_get_voltage(const struct device *dev, int32_t *volt_uv)
+{
+	const struct regulator_gpio_data *data = dev->data;
+
+	*volt_uv = data->current_volt_uv;
+	return 0;
+}
+
+static const struct regulator_driver_api regulator_gpio_api = {
+	.set_voltage = regulator_gpio_set_voltage,
+	.get_voltage = regulator_gpio_get_voltage,
+	.count_voltages = regulator_gpio_count_voltages,
+	.list_voltage = regulator_gpio_list_voltage,
+};
+
+static int regulator_gpio_init(const struct device *dev)
+{
+	const struct regulator_gpio_config *cfg = dev->config;
+	struct regulator_gpio_data *data = dev->data;
+	int ret;
+
+	regulator_common_data_init(dev);
+
+	if (!gpio_is_ready_dt(&cfg->switch_volt)) {
+		LOG_ERR("%s: gpio port: %s not ready", dev->name, cfg->switch_volt.port->name);
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&cfg->switch_volt,
+				    cfg->initial_state ? GPIO_OUTPUT_ACTIVE :
+							 GPIO_OUTPUT_INACTIVE);
+	if (ret < 0) {
+		LOG_ERR("%s: can't configure pin", dev->name);
+		return ret;
+	}
+
+	ret = regulator_gpio_get_state_voltage(dev, cfg->initial_state, &data->current_volt_uv);
+	if (ret < 0) {
+		LOG_ERR("%s: can't get list volatge for state %d", dev->name, cfg->initial_state);
+		return ret;
+	}
+
+	ret = regulator_common_init(dev, true);
+	if (ret < 0) {
+		LOG_ERR("%s: can't init common part of regulator", dev->name);
+		return ret;
+	}
+
+	return 0;
```

Comment:
```suggestion
	return regulator_common_init(dev, true);
```

why is `true` passed to the enable flag?

Reply:
fixed